src/langcheck/metrics/zh/reference_based_text_quality.py

```diff
@@ -0,0 +1,271 @@
+    is downloaded from HuggingFace and run locally. This is the default model
+    type and there is no setup needed to run this. this model will return cosine
+    similarity around 0.3 when sentences has no semantic similarity. sentences
+    with missing punctuation/different punctuation (one is declarative sentence,
+    one is question) would lower the value to 0.2 ~ 0.3.
+
+    2. The 'openai' type, where we use OpenAI's 'text-embedding-ada-002' model
+    by default (this is configurable). See
+    `this example <https://langcheck.readthedocs.io/en/latest/metrics.html
+    #computing-metrics-with-openai-models>`__
+    for examples on setting up the OpenAI API key.
+
+    Ref:
+        https://huggingface.co/tasks/sentence-similarity
+        https://www.sbert.net/docs/usage/semantic_textual_similarity.html
+        https://openai.com/blog/new-and-improved-embedding-model
+
+    Args:
+        generated_outputs: The model generated output(s) to evaluate
+        reference_outputs: The reference output(s)
+        prompts: The prompts used to generate the output(s). Prompts are
+            optional metadata and not used to calculate the metric.
+        embedding_model_type: The type of embedding model to use ('local' or
+            'openai'), default 'local'
+        openai_args: Dict of additional args to pass in to the
+            `openai.Embedding.create` function, default None
+
+    Returns:
+        An :class:`~langcheck.metrics.metric_value.MetricValue` object
+
+    .. note:: It seems same methods in different language share the same
+    pydoc, using sphinx or something to make these pydoc easier to be managed?
+    '''
+    generated_outputs, reference_outputs, prompts = validate_parameters_reference_based(  # NOQA: E501
+        generated_outputs, reference_outputs, prompts)
+    assert embedding_model_type in [
+        'local', 'openai'
+    ], ('Unsupported embedding model type. '
+        'The supported ones are ["local", "openai"]')
+
+    if embedding_model_type == 'openai':
+        # We can use the same API as english semantic_similarity to compare the
+        # similarity
+        metric_value = en_semantic_similarity(generated_outputs,
+                                              reference_outputs, prompts,
+                                              embedding_model_type, openai_args)
+        metric_value.language = 'zh'
+        return metric_value
+
+    # According to C-MTEB Benchmark
+    # https://github.com/FlagOpen/FlagEmbedding/tree/master/C_MTEB
+    # 3 different size model provided by BAAI is the Best 3 on embedding task
+    # Ref:
+    # https://huggingface.co/BAAI/bge-base-zh-v1.5
+    # using this model, it is hard to find two sentence cos_sim < 0.25.
+    # NOTE: as LLM generated content is more like paraphrase than senteces
+    # assests. there is no paraphrase2paraphrase benchmark yet, I think
+    # both of these models may not embedding paraphrase well.
```

Comment:
the datasets trained semantic-simlarity model include 'short text 2 short text',
But some peple add 'short text 2 pharphase' data in to training to imporve the model's ability when handling long text in 2022.

(shibing624/text2vec-base-chinese)[https://huggingface.co/shibing624/text2vec-base-chinese]
>shibing624/text2vec-base-chinese-paraphrase模型，是用CoSENT方法训练，基于nghuyong/ernie-3.0-base-zh用人工挑选后的中文STS数据集[shibing624/nli-zh-all/text2vec-base-chinese-paraphrase-dataset](https://huggingface.co/datasets/shibing624/nli-zh-all/tree/main/text2vec-base-chinese-paraphrase-dataset)，数据集相对于[shibing624/nli-zh-all/text2vec-base-chinese-sentence-dataset](https://huggingface.co/datasets/shibing624/nli-zh-all/tree/main/text2vec-base-chinese-sentence-dataset)加入了s2p(sentence to paraphrase)数据，**强化了其长文本的表征能力**(長い文章の対応能力を強化)，并在中文各NLI测试集评估达到SOTA(2022)

but the model opend by BAAI in 2023 do not mention they used it or not. So I note that the model may not give accurate score in long text understanding task.

Reply:
Got it, thanks for the explanation! In that case, I think it might be better to mark this as a TODO rather than a note, to make it clear that we're not sure how well it will perform and that we want to clarify this. I added my suggestion below

```suggestion
    # TODO: since the BAAI models may not have been trained on long texts (e.g.
    # it's unclear if it includes data such as
    # shibing624/nli-zh-all/text2vec-base-chinese-paraphrase-dataset in the
    # training data), it may not perform as well on longer texts. We should
    # benchmark this model's performance on longer text data.
```

Reply:
I found the paper [BAAI/C-MTEB](https://arxiv.org/pdf/2309.07597.pdf) here, In appendix "
C-MTP Composition" they list the training data, I checked some of them, they include long text in the training data. so, I think we can delete the TODO completely.

Reply:
awesome, thanks for the investigation! deleting the TODO sounds good